### ISSUE TITLE
V2 - Recode

### ISSUE THREAD

**Issue by CodeBrauer** [CodeBrauer/gmod-loadingscreen#22]
Because more and more people use this loadingscreen and I am having more free time, I decided the best part would be recoding this whole project and make it easier. Also the technologies will be new, because this time I want to keep it simple.

## Major changes

- [ ] Recode in JS
- [ ] Some example templates/layouts
- [ ] Easy setup & configuration

## Old features
- [ ] Image slideshow
- [ ] Music player
- [ ] Image slideshow
- [ ] Welcome message with steam-player name

## New features (roadmap)
- [ ] Loading progress

Share your ideas and feedback so v2.0 should get better and better!

**Comment by CodeBrauer** [comment]
New design draft: https://codepen.io/gab/pen/QMpOOK?editors=0100

**Comment by miguel456** [comment]
I have a feeling that V2 is going to be a lot better... Are you going to use PHP in V2?

**Comment by miguel456** [comment]
You could post the  roadmap in a Trello blog... I would be a good idea I think

**Comment by CodeBrauer** [comment]
@miguel456 - No, just pure Javascript. I have to make a small workaround for the Steam-WebAPI-call, but it should be doable

**Comment by CodeBrauer** [comment]
Demo is here: http://cdn.rawgit.com/CodeBrauer/gmod-loadingscreen/v2.0/index.html?steamid=76561198051267973&mapname=cs_office

But I have to test it on a real garrysmod server, to ensure the functions really work. Currently everything is just filled by `demo.js` with some demo data according to the [gmod wiki](https://wiki.garrysmod.com/page/Loading_URL)

**Comment by CodeBrauer** [comment]
Also the solution with anyorigin.com is not the best, I will look after a better solution.

**Comment by XproGrunds** [comment]
well im kinda sad it looks fine in Chrome but when i connect to my server the loading screen is  just broken

**Comment by CodeBrauer** [comment]
@XproGrunds Can you provide a screenshot?

**Comment by XproGrunds** [comment]
@CodeBrauer  well i dont know what i mest up here 

![sadface](https://user-images.githubusercontent.com/49123733/55423848-21ee8c80-557f-11e9-9d8c-9ddec68a28fe.PNG)

**Comment by XproGrunds** [comment]
@CodeBrauer Well i finaly got it so it doesn't look broken now i only have 3 things that is still broken the steam logo and the name is not comming up and the info and rules is cut in half

![plswork](https://user-images.githubusercontent.com/49123733/55500249-35195f00-5648-11e9-906e-7c1fe0f54fde.PNG)

**Comment by XproGrunds** [comment]
> @XproGrunds Can you provide a screenshot?

yes

**Comment by Torbikini** [comment]
Yeah, I'm having the same issue too. Not sure if it's just because I'm using the demo one, but something is off. The gradient is missing too.
I'm using this: `sv_loadingurl "https://cdn.rawgit.com/CodeBrauer/gmod-loadingscreen/v2.0/index.html?steamid=%s&mapname=%m"`
![image](https://user-images.githubusercontent.com/841055/56010500-0ed76d00-5ca1-11e9-9517-7c4ac7d721b3.png)
That's the outcome.

**Comment by CodeBrauer** [comment]
I've made some small changes, can you test it again?

I'm not anymore into gmod and have no time testing this screen. I just wanted to provide a free and open source good-looking screen.

**Comment by CodeBrauer** [comment]
Try this sv_loadingurl:

```
https://raw.githack.com/CodeBrauer/gmod-loadingscreen/302a401aeef537c24b55f8c44c2b6168db5b4960/index.html?steamid=%s&mapname=%m
```

**Comment by Torbikini** [comment]
Yeah, I really like your loading screen and think it's a lot better than the premium offerings. I really appreciate that you're still working on this despite not being interested in Garry's Mod anymore. I tried the URL above and it kinda works:
![image](https://user-images.githubusercontent.com/841055/56168924-d9c96400-5f99-11e9-950f-acce2348e90e.png)
The blue overlay is missing and it doesn't display properly.

**Comment by delwig** [comment]
I guess everyone having the same issue then? with the loading screen looking like the image above?

**Comment by XproGrunds** [comment]
@delwig have seen some servers that got it working fine but i wonder how they did it

**Comment by hofq** [comment]
I have the same Problem like @cookieman768 but i dont know how to repair it on my own Webspace, if any person knows how to do it pls let me know about it. I custumized this a lot and wont waste it.
